### PR TITLE
feat: deploy cheesechain, worldchain, xlayer

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description
- Support deployment to cheesechain, worldchain and zlayer

### Drive-by changes
- update `CosmosChainSchema` to add `chain_type` to fix breaking change introduced by https://github.com/cosmos/chain-registry/commit/6854133f987b577aae05d947719c7e04a6a44bbf

### CODE REVIEW

File: rust/hyperlane-core/src/chain.rs [hunk 1]
```diff
@@ -52,12 +52,14 @@ impl<'a> std::fmt::Display for ContractLocator<'a> {
...
     Ancient8 = 888888888,
     Arbitrum = 42161,
+    Arthera = 10242,
     Avalanche = 43114,
     #[cfg_attr(feature = "strum", strum(serialize = "bsc"))]
     BinanceSmartChain = 56,
     Blast = 81457,
     Bob = 60808,
     Celo = 42220,
+    Cheeschain = 383353,
```
Comment (daniel-savu) [inline rust/hyperlane-core/src/chain.rs:62]: small typo here
```suggestion
    Cheesechain = 383353,
```

File: rust/hyperlane-core/src/chain.rs [hunk 2]
```diff
@@ -52,12 +52,14 @@ impl<'a> std::fmt::Display for ContractLocator<'a> {
 pub enum KnownHyperlaneDomain {
     Ancient8 = 888888888,
     Arbitrum = 42161,
+    Arthera = 10242,
```
Comment (tkporter) [inline rust/hyperlane-core/src/chain.rs:55]: we can rm Arthera from this PR and just wait to apply the on-chain config to remove it

File: .registryrc [hunk 3]
```diff
@@ -1 +1 @@
-7c6bfefbaefb6aad19b114afb6743b1b70c69552
+179d801b80ec36b6dd54cb8adcf25e1c9e50db97
```
Comment (Mo-Hussain) [inline .registryrc:1]: @tkporter - we should set this to the commit after we merge the corresponding registry PR?

File: rust/config/mainnet_config.json [hunk 4]
```diff
@@ -131,6 +132,67 @@
       "timelockController": "0x0000000000000000000000000000000000000000",
       "validatorAnnounce": "0x1df063280C4166AF9a725e3828b4dAC6c7113B08"
     },
+    "arthera": {
```
Comment (Mo-Hussain) [inline rust/config/mainnet_config.json:135]: Do we want to remove references to arthera for generated files like this one now @tkporter

File: rust/hyperlane-core/src/chain.rs (hunk 2 above)
Comment (Mo-Hussain) [reply rust/hyperlane-core/src/chain.rs]: Do we want to update the image tags in this PR now that we've removed references to Arthera?

File: .registryrc (hunk 3 above)
Comment (tkporter) [reply .registryrc:1]: yep!

File: typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json [hunk 5]
```diff
@@ -1,6 +1,8 @@
 {
   "ancient8": {
-    "validators": ["0xbb5842ae0e05215b53df4787a29144efb7e67551"]
+    "validators": [
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json:3]: might need to `yarn prettier` - it's annoying but prettier fights the address serialization of the tooling

File: typescript/infra/config/environments/mainnet3/agent.ts [hunk 6]
```diff
@@ -155,12 +160,15 @@ export const hyperlaneContextAgentChainConfig: AgentChainConfig<
     polygonzkevm: true,
     redstone: true,
     // Out of caution around pointer contracts (https://www.docs.sei.io/dev-interoperability/pointer-contracts) not being compatible
-    // and the scraper not gracefully handling txs that may not exist via the eth RPC, we don't run the scraper.
+    // and the scraper
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/agent.ts:163]: nit: I assume this was a linter / prettier thing? if not, i guess revert? but if so might be nicer to have this line be as long as possible and have the short one be the final line

File: typescript/sdk/src/utils/cosmos.ts [hunk 7]
```diff
@@ -9,6 +9,7 @@ export const CosmosChainSchema = z
       .regex(new RegExp('^(\\.\\./)+chain\\.schema\\.json$'))
       .optional(),
     chain_name: z.string().regex(new RegExp('[a-z0-9]+')),
+    chain_type: z.string().regex(new RegExp('[a-z0-9]+')),
```
Comment (tkporter) [inline typescript/sdk/src/utils/cosmos.ts:12]: dang so zod complained when there was an extra field that was supplied without the schema knowing about it? very annoying. Wonder if there's a way to be tolerant of the data being a superset of the type as that feels safe for our purposes here

**Review by tkporter — APPROVED** [review]

File: rust/hyperlane-core/src/chain.rs (hunk 2 above)
Comment (tkporter) [reply rust/hyperlane-core/src/chain.rs]: Nah that's okay. Their presence in the agents won't have any negative impact

File: typescript/sdk/src/utils/cosmos.ts (hunk 7 above)
Comment (Mo-Hussain) [reply typescript/sdk/src/utils/cosmos.ts:12]: That makes sense, will investigate

File: typescript/infra/config/environments/mainnet3/agent.ts (hunk 6 above)
Comment (Mo-Hussain) [reply typescript/infra/config/environments/mainnet3/agent.ts:163]: Not sure how this happened, will revert

File: typescript/sdk/src/utils/cosmos.ts [hunk 8]
```diff
@@ -442,7 +442,7 @@ export const CosmosChainSchema = z
     keywords: z.array(z.string()).optional(),
     extra_codecs: z.array(z.enum(['ethermint', 'injective'])).optional(),
   })
-  .strict()
+  .passthrough()
```
Comment (Mo-Hussain) [inline typescript/sdk/src/utils/cosmos.ts:445]: @tkporter - this did the trick

**Pull request merged**